src/wallet/rpcwallet.cpp

```diff
@@ -537,6 +538,89 @@ static RPCHelpMan sendtoaddress()
+            "                   prevailing fee market. Must be one of (case insensitive):\n"
+            "\"" + FeeModes("\"\n\"") + "\""},
+                },
+                RPCResult{
+                    RPCResult::Type::OBJ, "", "",
+                    {
+                        {RPCResult::Type::NUM, "feerate", /* optional */ true, "estimate fee rate in " + CURRENCY_UNIT + "/kvB (only present if no errors were encountered)"},
+                        {RPCResult::Type::STR, "reason", /* optional */ true, "mention the reason for the estimated fee"},
+                        {RPCResult::Type::ARR, "errors", /* optional */ true, "Errors encountered during processing (if there are any)",
+                            {
+                                {RPCResult::Type::STR, "", "error"},
+                            }},
+                        {RPCResult::Type::NUM, "blocks", "block number where estimate was found (only when reason is estimatesmartfee)\n"
+                        "The request target will be clamped between 2 and the highest target\n"
+                        "fee estimation is able to return based on how long it has been running.\n"
+                        "An error is returned if not enough transactions and blocks\n"
+                        "have been observed to make an estimate for any number of blocks."},
+                    }},
+                RPCExamples{
+                    HelpExampleCli("walletestimatefee", "6")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::shared_ptr<CWallet> const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!pwallet) return NullUniValue;
+
+    // Make sure the results are valid at least up to the most recent block
+    // the user could have gotten from another RPC command prior to now
+    pwallet->BlockUntilSyncedToCurrentChain();
+
+    LOCK(pwallet->cs_wallet);
+
+    // Wallet comments
+    CCoinControl coin_control;
+
+    //dummy variable temp representing NULL fee_rate
+    UniValue temp(UniValue::VNULL);
+
+    SetFeeEstimateMode(*pwallet, coin_control, /* conf_target */ request.params[0], /* estimate_mode */ request.params[1], /* fee_rate */temp , /* override_min_fee */ false);
+
+    UniValue result(UniValue::VOBJ);
+    UniValue errors(UniValue::VARR);
+    UniValue reas(UniValue::VSTR);
+    FeeCalculation feeCalc;
+    CFeeRate effective_feerate = GetMinimumFeeRate(*pwallet, coin_control, &feeCalc);
+
+    if (effective_feerate != CFeeRate(0)) {
+        result.pushKV("feerate", ValueFromAmount(effective_feerate.GetFeePerK()));
+        //when reason is estimatesmartfee return block number where estimate was found
+        if(feeCalc.reason!=FeeReason::MEMPOOL_MIN&&
+           feeCalc.reason!=FeeReason::REQUIRED&&
+           feeCalc.reason!=FeeReason::FALLBACK&&
+           feeCalc.reason!=FeeReason::PAYTXFEE)
+        {
```

Comment:
general preference is for brackets { to be on the same line as the conditional, no big deal though